File: project_trademonitor/order_collection/src/main/java/com/gupao/bd/trademonitor/dcs/dao/HBaseClient.java
package com.gupao.bd.trademonitor.dcs.dao;

import com.alibaba.fastjson.JSONObject;
import org.apache.hadoop.conf.Configuration;
import org.apache.hadoop.hbase.HBaseConfiguration;
import org.apache.hadoop.hbase.TableName;
import org.apache.hadoop.hbase.client.Connection;
import org.apache.hadoop.hbase.client.ConnectionFactory;
import org.apache.hadoop.hbase.client.Put;
import org.apache.hadoop.hbase.client.Table;

import java.io.IOException;
import java.util.Map;

/**
 * 功能：实现HBase的crud操作
 */
public class HBaseClient {

    private Connection conn;

    public HBaseClient(String zkQuorum) {
        Configuration defaultConf = HBaseConfiguration.create();
        defaultConf.set("hbase.zookeeper.quorum", zkQuorum);

        try {
            conn = ConnectionFactory.createConnection(defaultConf);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public void put(String tableName, String rowkey, JSONObject record) {
        try {
            Table table = conn.getTable(TableName.valueOf(tableName));

            Put put = new Put(rowkey.getBytes());

            for (Map.Entry<String, Object> entry : record.entrySet()) {
                put.addColumn("info".getBytes(), entry.getKey().getBytes(), entry.getValue().toString().getBytes());
            }

            table.put(put);
            table.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: project_trademonitor/metric_compute/src/main/scala/com/gupao/bd/trademonitor/mcs/dao/HBaseClient.scala
package com.gupao.bd.trademonitor.mcs.dao

import java.util.{ArrayList => jArrayList}

import org.apache.hadoop.hbase.client._
import org.apache.hadoop.hbase.{HBaseConfiguration, TableName}

/**
  * 功能：从HBase加载/写入数据
  **/
class HBaseClient(zkQuorum: String) {

  val conf = HBaseConfiguration.create()
  conf.set("hbase.zookeeper.quorum", zkQuorum)

  val conn: Connection = ConnectionFactory.createConnection(conf)

  val COL_FAMILY = "info".getBytes

  def get(tableName: String, rowkey: String): Result = {
    val table = conn.getTable(TableName.valueOf(tableName))
    val get: Get = new Get(rowkey.getBytes)
    val result: Result = table.get(get)
    result
  }
}